Description: enable emulator check
Author: Dmitrijs Ledkovs <xnox@ubuntu.com>

--- android-tools-4.2.2+git20130218.orig/core/adbd/transport_local.c
+++ android-tools-4.2.2+git20130218/core/adbd/transport_local.c
@@ -309,15 +309,16 @@ void local_init(int port)
 #else
         /* For the adbd daemon in the system image we need to distinguish
          * between the device, and the emulator. */
-        char is_qemu[PROPERTY_VALUE_MAX];
+        //char is_qemu[PROPERTY_VALUE_MAX];
         //property_get("ro.kernel.qemu", is_qemu, "");
         //if (!strcmp(is_qemu, "1")) {
+        if (access("/dev/qemu_pipe", R_OK|W_OK) != -1) {
             /* Running inside the emulator: use QEMUD pipe as the transport. */
-        //    func = qemu_socket_thread;
-        //} else {
+            func = qemu_socket_thread;
+        } else {
             /* Running inside the device: use TCP socket as the transport. */
             func = server_socket_thread;
-            //}
+        }
 #endif // !ADB_HOST
     }
 
